### ISSUE TITLE
State of integration of Recording API

### ISSUE THREAD

**Issue by shimonomihs** [microg/GmsCore#2690]
Hello,
what is the state of the new Recording API, that replaces the Google fit API.
https://developer.android.com/health-and-fitness/guides/recording-api
An application that relies on this api for step recording when it is closed does not in fact record steps when it is closed.

**Comment by Klemet** [comment]
As mentioned in #1700 , I have an app that seems to hang indefinitely because of this issue. I'm not certain though, so keep this with a grain of salt.

The app is called [Walkscape](https://walkscape.app/); it's an upcoming Android/iOS game that is currently in closed Beta. It's basically Runescape, except that you have to walk to do actions/travel/grind.

The game relies on `com.google.android.gms:play-services-fitness` to obtain the step counts that happened when the game was closed/off focus. So while the game does record steps correctly as long as it stays opened and active on the screen, it hangs when trying to retrieve the steps when being restarted or switched back to. Just hanging on a "Loading steps..." message, not crashing.

Talking with the dev team of Walkscape, some of them suspect that the problem lies in my lack of this google library. It seems like there is currently a dummy API to interact with `play-services-fitness` from MicroG now (in [v0.3.7.250932](https://github.com/microg/GmsCore/releases/tag/v0.3.7.250932) which I have installed manually to be sure), but I imaging that my problem is similar to Shimon here (recording steps when an app is closed).

**Comment by Klemet** [comment]
(I'll add that I seem to currently be the only beta-tester of Walkscape with this particular issue, and that people that had the same issue in the past fixed it by finally installing `com.google.android.gms:play-services-fitness` on their phones through the sandbox mode of Graphene OS. So while I can't be 100% certain, it looks pretty certain that the issue lies here in my case).

**Comment by Idcrafter** [comment]
isn't it soon not needed because google is deprecating it in 2026 as now stated on the api site you linked?

**Comment by Klemet** [comment]
Yes, it make sense ! Now, it seems to have been replace by google "[Health Connect](https://play.google.com/store/apps/details?id=com.google.android.apps.healthdata)" ? Not certain of this 🤔.

In any case, I was able to install Health Connect; not ideal since it's from Google, but I was able to cut its internet connection using AfWall +. Now, everything works ! But I'd be happy if a MicroG alternatives is made one day 😁


Thank you again for all of your work !

**Comment by Idcrafter** [comment]
Isn't Health connect inside AOSP since like atleast Android 15, could be sooner? They improved it to being able to count steps in Android 16 QPR2 and with it have no need to be replaced by microG for those devices and as it is a default user app on older Android Devices which means that every devs could make a app for it to not also burden the microG team with that. 
I find it better to have it as a open api system wide instead of fragmented api's but idk about if it needs internet when it is the version shipping in AOSP tho, you can block internet on lineageOS tho so is that not a huge problem i would think.
